elasticsearch/src/main/java/org/hibernate/search/elasticsearch/client/impl/ArbitrarySort.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.elasticsearch.client.impl;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.hibernate.search.elasticsearch.impl.GsonHolder;
+
+import io.searchbox.core.search.sort.Sort;
+
+/**
+ * Implementation of a sort with arbitrary keys and values for Jest.
+ * <p>This is useful for providing a way for users to use sort features
+ * that are not otherwise implemented in Hibernate Search.
+ *
+ * @author Yoann Rodiere
+ */
+public class ArbitrarySort extends Sort {
+
+	private String fieldName;
+
+	private Object sortDescription;
+
+	public ArbitrarySort(String fieldName, String sortDescription) {
+		super( fieldName );
```

Comment:
@emmanuelbernard Guillaume is concerned about "byNative(String fieldName, String sortSpec)" requiring two arguments, because in some case the `fieldName` argument isn't really a field name, but more an Elasticsearch keyword (such is the case for `_geo_location` and `_script`).
In short, I say it's easier to use that way and seems consistent with the way to do in ElasticSearch, but Guillaume thinks it's hackish.

On second though, depending on how we want to make the DSL generic later, having two arguments instead of one may be problematic. I don't know what was discussed in Rome exactly about the DSL generification, so...